### ISSUE TITLE
upload issue with one of the deposit

### ISSUE THREAD

**Issue by llehtine** [EUDAT-B2SHARE/b2share#834]
B2Share deposit 310 (https://b2share.eudat.eu/record/310) seems to have the other file deposited with the temporary file name, which are used internally by the file upload.

In temp directory the file didn't have the metadata-file:

Draft_1.txt_469075763564851730c0a35f9d3123bf.txt
Draft.txt_5dc41103d7209b8f02e592570afa1800_0
metadata_Draft_1.txt_469075763564851730c0a35f9d3123bf.txt

On the log files it seems the the uploading was never completed:
[Wed Jan 27 23:17:43 2016] [error] INFO:invenio.base:uploaded chunk Draft.txt_5dc41103d7209b8f02e592570afa1800_0
[Wed Jan 27 23:17:43 2016] [error] INFO:invenio.base:uploaded chunk Draft_1.txt_469075763564851730c0a35f9d3123bf_0
[Wed Jan 27 23:17:43 2016] [error] INFO:invenio.base:finished uploading: Draft_1.txt, size 4, in /invenioupload/ss/4924cc06c53b11e5a9f5005056be6c1b/Draft_1.txt_469075763564851730c0a35f9d3123bf.txt

**Comment by llehtine** [comment]
Finally had some time to play with this. I was able to replicate this issue two times on test-env, with these steps:
1) Create new deposit, add one file, and upload it
2) Select second file, which would upload as chunks (I've used 200MB-file), and start uploading it,
but somewhere in the middle stop
3) Delete the second file from deposit
4) Complete the deposit

Chunks of the incompleted file are still inside temporary upload-folder, but without metadata-file. When the deposit is completed, get_depositing_files_metadata file go through all the files in the temporary directory, regardless if they have valid metadata-file (it will give error to logs and use default metadata if medata-file is missing, but still will take files to deposition)

https://test-eudatis.csc.fi/record/301
https://test-eudatis.csc.fi/record/302

I guess the problem here is in b2share_upload_handler.py
When deleting file, it will take filename-parameter from the form:
``` filename = request.form['filename']```

but when there is multiple-files, it will take only the first filename from the deposit, and check that. So for the second, third etc. files which are uploaded, temporary files are never deleted from the filesystem, even if the file is deleted from the deposit.
